src/Administration/Resources/app/administration/src/module/sw-order/component/sw-order-promotion-field/index.js

```diff
@@ -248,6 +252,18 @@ export default {
             }
         },
 
+        /**
+         * To prevent losing unsaved changes on recalculate the order,
+         * we need to save the **versioned** order beforehand.
+         */
+        async saveAndRecalculate(afterRecalculateFn = null) {
+            if (this.swOrderDetailOnSaveAndRecalculate) {
+                await this.swOrderDetailOnSaveAndRecalculate(afterRecalculateFn);
+            } else {
+                this.$emit('save-and-recalculate', afterRecalculateFn);
+            }
+        },
```

Comment:
saveAndRecalculate doesn't have a afterFn:
```suggestion
        async saveAndRecalculate() {
            if (this.swOrderDetailOnSaveAndRecalculate) {
                await this.swOrderDetailOnSaveAndRecalculate();
            } else {
                this.$emit('save-and-recalculate');
            }
        },
```